src/cargo/ops/cargo_compile.rs

```diff
@@ -182,7 +183,7 @@ pub fn compile_pkg<'a>(root_package: &Package,
         Some(args) => {
             if to_builds.len() == 1 {
                 let targets = try!(generate_targets(to_builds[0], profiles,
-                                                    mode, filter, release));
+                                                    mode, filter, all_feature, release));
```

Comment:
Ah yeah the `resolve.features(pkgid)` is how the set of activated features is learned about. So by moving the `all_feature` variable down here it'll use the right package to learn the set of activated features.